korangar/src/audio/audio_state.rs

```diff
@@ -0,0 +1,40 @@
+use std::sync::Arc;
+
+use korangar_audio::AudioEngine;
+
+use crate::GameFileLoader;
+pub struct AudioWindowState {

```

Comment:
I don't think you need this state at all. You can just call the audio engine functions directly.

Reply:
I think it is better to create a struct that handles the audio settings states but not sure how to call the function and to not directly put this audio function inside the main loop.

Reply:
But look how much it currently does (next to nothing). So this abstraction is not needed yet!